webapp/src/components/training/Training.vue

```diff
@@ -38,22 +38,42 @@ import Finished from '@/components/training/Finished.vue'
 import Data from '@/components/data/Data.vue'
 
 const router = useRouter()
+const route = useRoute()
```

Comment:
same as above? I think you can merge both

Reply:
They are different objects!
From the [vue-router doc](https://router.vuejs.org/guide/): "The key thing to notice is that the composables useRouter() and useRoute() are used to access the router instance and current route respectively."